File: nebras-app/app/src/main/kotlin/com/nebras/mobile/core/media/ContentContext.kt
package com.nebras.mobile.core.media

/**
 * سياق تشغيل المحتوى (Content Context).
 *
 * يُستخدم من قِبَل مُشغّلات الصوت والفيديو ليفهما "من أين جاء المستخدم":
 * - [FOR_YOU] : من صفحة "لك" (موجز التوصيات) — نُفعِّل Smart Auto-Play.
 * - [SEARCH]  : من نتائج البحث — نُفعِّل Smart Auto-Play أيضاً.
 * - [SECTION] : من تصفّح قسم (قائمة تشغيل مبنيّة مسبقاً) — لا نحتاج auto-fill.
 * - [SAVED]   : من المحفوظات — نلتزم بالقائمة ولا نُضيف عشوائيّاً.
 * - [HOME]    : من الشاشة الرئيسيّة — Smart Auto-Play بسياسة متحفّظة.
 * - [DIRECT]  : فتح مباشر (إشعار/رابط) — سلوك "مقطع واحد" ثمّ انتهاء.
 *
 * ⚠ عند وجود قائمة تشغيل حقيقيّة نتنقّل داخلها أولاً (Next/Prev). فقط إذا
 * لم يتبقَّ عنصر تالٍ نُطلق `onExhaustedAutoFill` — ولا يحدث ذلك إلّا في
 * السياقات التي تسمح به (انظر [allowsSmartAutoPlay]).
 */
enum class ContentContext {
    FOR_YOU,
    SEARCH,
    SECTION,
    SAVED,
    HOME,
    DIRECT,
    ;

    /**
     * هل يسمح هذا السياق بتشغيل اقتراحات ذكيّة عند نفاد القائمة؟
     * القاعدة: نعم في "لك / بحث / رئيسية"، ولا في "قسم / محفوظ / مباشر".
     */
    val allowsSmartAutoPlay: Boolean
        get() = when (this) {
            FOR_YOU, SEARCH, HOME -> true
            SECTION, SAVED, DIRECT -> false
        }
}

File: app-flutter/lib/core/media/content_context.dart
/// سياق تشغيل المحتوى (Content Context).
///
/// يُستخدم من قِبَل مُشغّلات الصوت والفيديو ليفهما "من أين جاء المستخدم":
/// - [forYou]    : من صفحة "لك" (موجز التوصيات) — نُفعِّل Smart Auto-Play.
/// - [search]    : من نتائج البحث — نُفعِّل Smart Auto-Play أيضًا.
/// - [section]   : من تصفّح قسم (قائمة تشغيل مبنيّة مسبقًا) — لا نحتاج auto-fill.
/// - [saved]     : من المحفوظات — نلتزم بقائمة المحفوظات ولا نُضيف عشوائيًّا.
/// - [home]      : من الشاشة الرئيسيّة — نُفعِّل Smart Auto-Play بسياسة متحفّظة.
/// - [direct]    : فتح مباشر (إشعار/رابط) — سلوك "مقطع واحد" ثمّ انتهاء.
///
/// ⚠ عند وجود قائمة تشغيل حقيقيّة نتنقّل داخلها أولًا (Next/Prev).
/// فقط إذا لم يتبقَّ عنصر تالٍ، نُطلق `onExhaustedAutoFill` — وهذا
/// لا يحدث إلّا في السياقات التي تسمح به (انظر [allowsSmartAutoPlay]).
enum ContentContext {
  forYou,
  search,
  section,
  saved,
  home,
  direct,
}

extension ContentContextX on ContentContext {
  /// هل يسمح هذا السياق بتشغيل اقتراحات ذكيّة عند نفاد القائمة؟
  /// القاعدة: نعم في "لك / بحث / رئيسية"، ولا في "قسم / محفوظ / مباشر".
  bool get allowsSmartAutoPlay {
    switch (this) {
      case ContentContext.forYou:
      case ContentContext.search:
      case ContentContext.home:
        return true;
      case ContentContext.section:
      case ContentContext.saved:
      case ContentContext.direct:
        return false;
    }
  }
}
